hail/src/main/scala/is/hail/expr/types/physical/PSet.scala

```diff
@@ -8,14 +8,10 @@ import org.json4s.jackson.JsonMethods
 
 import scala.reflect.{ClassTag, _}
 
-final case class PSet(elementType: PType, override val required: Boolean = false) extends PContainer {
+final case class PSet(elementType: PType, override val required: Boolean = false) extends PIterable {
```

Comment:
don't change this here.

I'm quite strongly opposed to reorganizational PRs that also make semantic changes like this -- the large diffs mean that each line is reviewed less stringently, so unintended consequences of changes like this are more likely to leak through.

As I commented above, going to fundamental type to access container methods is not an option.

Something you can do to make this change incremental is give PSet and PDict a class variable:
```scala
val arrayRep: PArray = PCanonicalArray(elementType, required)
```

Then implement container methods using that.

Reply:
If I continue to inherit from PContainer, then PSet and PDict will need to implement all PContainer methods. What should the implementations for these be? Currently to keep functionality identical I would copy and paste the implementation in PCanonicalArray.

Reply:
as above, this:
```scala
val arrayRep: PArray = PCanonicalArray(elementType, required)

def loadElement(...) = arrayRep.loadEleemnt(...)
```

Reply:
I see; I had chosen not to do this to avoid duplicating the PCanonicalArray implementation. Will follow this strategy.

Reply:
Man, there are a lot of PContainer methods. You can avoid code duplication between PSet and PDict like so:
```scala
abstract class PArrayBackedContainer(arrayRep: PArray) extends PContainer {
  ... all the defs here ...
}

class PSet(...) extends PArrayBackedContainer(PCanonicalArray(eltType, required))
```

and eventually when PSet is an interface:
```scala
class PCanonicalSet(...) extends PArrayBackedContainer(PCanonicalArray(eltType, required)) with PSet
```